scylla/src/transport/connection.rs

```diff
@@ -304,6 +304,7 @@ impl Connection {
         if let Some(tracing_id) = query_response.tracing_id {
             prepared_statement.prepare_tracing_ids.push(tracing_id);
         }
+        prepared_statement.inherit_config(query.config.clone());
```

Comment:
Now that I gave it more thought, I don't think this is the best way to do it. We are creating a PreparedStatement on line `291` with the default StatementConfig and then we are overwriting it. Why not create a PreparedStatement with a correct config in the first place?

I suggest adding a new static method `PreparedStatement::new_with_config` which would be the same as `PreparedStatement::new` but would get an additional `StatementConfig` parameter. The new method would also have `pub(crate)` visibility.

Ideally, I'd like the `PreparedStatement::new` method to take a `StatementConfig` and make it `pub(crate)` - I don't think it makes sense for users to use it - but that would be a breaking change...

Reply:
I agree. And breaking is fine at this stage, we can always restore the visibility if somebody insists - manually creating a prepared statement as a user makes little sense though

Reply:
done